codegen/src/main/scala-2.13+/org/apache/pekko/grpc/gen/CodeGenerator.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * license agreements; and to You under the Apache License, version 2.0:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This file is part of the Apache Pekko project, which was derived from Akka.
+ */
+
+/*
+ * Copyright (C) 2018-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package org.apache.pekko.grpc.gen
+
+import com.google.protobuf.ExtensionRegistry
+import com.google.protobuf.compiler.PluginProtos.CodeGeneratorRequest
+import com.google.protobuf.compiler.PluginProtos.CodeGeneratorResponse
+import protocbridge.Artifact
+
+/**
+ * Code generator trait that is not directly bound to scala-pb or protoc (other than the types).
+ */
+trait CodeGenerator {
+  import CodeGenerator._
+
+  /** Generator name; example: `pekko-grpc-scala` */
+  def name: String
+
+  def run(request: CodeGeneratorRequest, logger: Logger): CodeGeneratorResponse
+
+  /** Takes Scala binary version and returns suggested dependency Seq */
+  def suggestedDependencies: ScalaBinaryVersion => Seq[Artifact]
+
+  def registerExtensions(registry: ExtensionRegistry): Unit = {}
```

Comment:
Add sclaladoc too？

Reply:
the original code (still in use for scala 2.12 compile) does not have scaladoc so I don't want to make the 2 versions differ from each other

Reply:
okay, but you can add doc in both place :)

Reply:
I'm not a fan of hijacking PRs to make unrelated changes - if you want new scaladoc, create a new Issue or PR